oak_containers_orchestrator/src/container_runtime.rs

```diff
@@ -0,0 +1,114 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Representation of a minimal OCI filesystem bundle config, including just the required fields
+/// Ref: <https://github.com/opencontainers/runtime-spec/blob/467fd17d4f2a987fa00051ce44b69bf9620e2ee6/config.md>
+#[derive(serde::Deserialize)]
+struct OciFilesystemBundleConfig {
+    process: OciFilesystemBundleConfigProcess,
+    root: OciFilesystemBundleConfigRoot,
+}
+
+#[derive(serde::Deserialize)]
+struct OciFilesystemBundleConfigProcess {
+    user: OciFilesystemBundleConfigProcessUser,
+    args: Vec<String>,
+}
+
+#[derive(serde::Deserialize)]
+struct OciFilesystemBundleConfigProcessUser {
+    uid: u32,
+    gid: u32,
+}
+
+#[derive(serde::Deserialize)]
+struct OciFilesystemBundleConfigRoot {
+    path: std::path::PathBuf,
+}
+
+async fn run_command_and_log_output(
+    command: &mut tokio::process::Command,
+) -> Result<(), Box<std::io::Error>> {
+    let output = command.output().await;
+    log::info!("{:?}: {:?}", command, output);
+    Ok(())
+}
+
+// Directory at which the container OCI filesystem bundle will be unpacked
+const CONTAINER_DIR: &str = "/oak_container";
+
+pub async fn run(container_bundle: &[u8]) -> Result<(), anyhow::Error> {
+    tokio::fs::create_dir(CONTAINER_DIR).await?;
+    tar::Archive::new(container_bundle).unpack(CONTAINER_DIR)?;
+
+    let oci_filesystem_bundle_config: OciFilesystemBundleConfig = {
+        let file_path = {
+            let mut base = std::path::PathBuf::from(CONTAINER_DIR);
+            base.push("config.json");
+            base
+        };
+        let oci_filesystem_bundle_config_file = tokio::fs::read_to_string(file_path).await?;
+
+        serde_json::from_str(&oci_filesystem_bundle_config_file)?
+    };
+
+    // mount host vsock device inside the container directory to allow the
+    // trusted application to communicate with the outside
+    {
+        run_command_and_log_output(
+            tokio::process::Command::new("rm")
```

Comment:
This feels a bit like a glorified shell script (just invoking `rm`, `mkdir`, etc...)

My personal preference would be to use proper Rust alternatives, like `std::fs::remove_dir_all` and friends.
(`mount` is probably the most difficult as you need to explicitly include the raw syscall crate as we do in stage1.)

Reply:
fair, i can follow up with another PR to remove the shell intermediary. It was just the fastest way to get things started. 

Can I use the nix crate? We technically don't need darwin support, but the interface of it is a lot more streamlined.

Reply:
The nix crate is fine. There is also some sample code for using it to mount pseudo-filesystems in https://github.com/project-oak/oak/blob/main/oak_echo_linux_init/src/init.rs